src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -359,38 +370,78 @@ public void onOpened()
         createXpBar();
         createSkillContent();
 
-        resourceList.setDataProvider(() -> getOpenRequestsOfCitizen().size(), (index, rowPane) ->
-        {
-            final ImmutableList<IRequest> openRequests = getOpenRequestsOfCitizen();
-            if (index < 0 || index >= openRequests.size())
-            {
-                return;
-            }
+        resourceList.setDataProvider(new ScrollingList.DataProvider() {
 
-            final IRequest request = openRequests.get(index);
-            final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
-            final List<ItemStack> displayStacks = request.getDisplayStacks();
+            private List<RequestWrapper> requestWrappers = null;
 
-            if (!displayStacks.isEmpty())
-            {
-                exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
-            }
-            else
+            @Override
+            public int getElementCount()
             {
-                final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
-                logo.setVisible(true);
-                logo.setImage(request.getDisplayIcon());
+                requestWrappers = getOpenRequestTreeOfCitizen();
+                return requestWrappers.size();
             }
 
-            rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(request.getToken()).getFormattedText());
-            rowPane.findPaneOfTypeByID(REQUEST_SHORT_DETAIL, Label.class)
-              .setLabelText(request.getShortDisplayString().getFormattedText().replace("§f", ""));
-
-            if (!(request.getRequest() instanceof IDeliverable)
-                  || (!isCreative && !InventoryUtils.hasItemInItemHandler(new InvWrapper(inventory),
-              stack -> ((IRequest<? extends IDeliverable>) request).getRequest().matches(stack))))
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
             {
-                rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
+                if (index < 0 || index >= requestWrappers.size())
+                {
+                    return;
+                }
+
+                final RequestWrapper wrapper = requestWrappers.get(index);
+                final Box wrapperBox = rowPane.findPaneOfTypeByID(WINDOW_ID_REQUEST_BOX, Box.class);
+                wrapperBox.setPosition(wrapperBox.getX() + 10*wrapper.getDepth(), wrapperBox.getY());
+                wrapperBox.setSize(wrapperBox.getParent().getWidth() - 10*wrapper.getDepth(), wrapperBox.getHeight());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)